extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/EngineProducer.java

```diff
@@ -262,6 +264,17 @@ public void run() {
         Qute.setEngine(engine);
     }
 
+    private void registerCustomLocators(EngineBuilder builder) {
```

Comment:
In fact, it would be a bit cleaner to inject `@All List<TemplateLocator> locators` in the constructor and then just use the list directly. And the reason is that `@Dependent` locator instances would be correctly bound to the lifecycle of the `EngineProducer`.

Reply:
Make sense, thank you. I had to use `@Any` though (didn't work without it).

Reply:
That's odd. `@All List` should be an equivalent of `@Inject @Any Instance`... does any of the tests fail if I remove `@Any`?

Reply:
If you remove `@Any`, run `mvn clean install -DskipTests -DskipITs` in runtime module, then `mvn clean verify -Dtest=CustomTemplateLocatorTest,BlankLocateValueTest,WrongLocationClassTargetTest` in deployment module, you will get 
```
Caused by: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type java.util.List<io.quarkus.qute.TemplateLocator> and qualifiers [@Identified(value = "2e58db6f0e5b3700ac655f4187711a407522b003"), @All]
```
Works with `@Any`.

Reply:
That looks like a bug. I'll try it out...

Reply:
https://github.com/quarkusio/quarkus/pull/25784

Reply:
I'll drop @Any and rebase the PR to the latest main once #25784 is merged.

Reply:
@Any dropped and pushed to the PR. Tests are passing, thank you